### PR TITLE
Refer OPTUNA_STORAGE environment variable in optuna CLI

### PR DESCRIPTION
**Pull request by Hakuyume**
## Motivation

Sometimes, storage URL contains sensitive information (e.g. DB password). It is dangerous to specify such URL in command line argument. This PR allows users to specify URL via environment variable.

### CODE REVIEW

File: tests/test_cli.py [hunk 1]
```diff
@@ -135,6 +135,27 @@ def test_create_study_command_without_storage_url() -> None:
     assert usage.startswith("usage:")
 
 
+@pytest.mark.skip_coverage
+def test_create_study_command_with_storage_env() -> None:
```
Comment (Hakuyume) [inline tests/test_cli.py:139]: copied from `test_create_study_command` except `check_call`

**Review by c-bata — APPROVED** [review]
LGTM! Thank you for your contribution.

**Comment by HideakiImamura** [comment]
@toshihikoyanase Could you review this PR?

**Review by toshihikoyanase — APPROVED** [review]
LGTM! Thank you.

I found some follow-up tasks. They shouldn't be resolved in this PR, but I'd like to leave them as future development items.
- If users set `OPTUNA_STORAGE`, some test cases including `test_create_study_command_without_storage_url`, `test_storage_upgrade_command`, and `test_delete_study_command_without_storage_url` in `tests/test_cli.py` will fail.
- `optuna` CLI follows Python API, so `optuna.get_storage` may be able to read the `OPTUNA_STORAGE` environment variable. But, such a feature is controversial.

**Pull request merged**